x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/common_model/ecs_event.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IEvent as IEventLogEvent } from '../../../../../../event_log/server';
+
+// https://www.elastic.co/guide/en/ecs/1.9/ecs-guidelines.html
+// https://www.elastic.co/guide/en/ecs/1.9/ecs-category-field-values-reference.html
+// https://www.elastic.co/guide/en/ecs/1.9/ecs-field-reference.html
+
+export type IEcsEvent = IEventLogEvent & IEcsAdditionalFields;
+
+interface IEcsAdditionalFields {
```

Comment:
I don't think these will actually be written out today, till they're added to our schema - we use `dynamic: false` at the top of the mappings, so I believe they will not get indexed, at the very least.

https://github.com/elastic/kibana/blob/6264c563d17bc72127adb200e06b3634a399463d/x-pack/plugins/event_log/generated/mappings.json#L1-L3

If this PR is just an experiment, you might want to extend the schema, following the directions here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/event_log/generated/README.md - and let us know if any of it is confusing so we can correct it.

I think we'd want to update the event log separately though, if we intend to merge this, just to keep the PRs a little cleaner.

Reply:
Yeah, this PR is just an experiment, a proposal. https://github.com/elastic/kibana/pull/94143#discussion_r592343636
I'll try to explain it better is a separate comment below.